featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -1116,43 +1159,34 @@ def can_stack_primitive_on_inputs(primitive, inputs):
         # However, we want to check stacking rules with the primitive the DirectFeature is based on.
         f_primitive = _find_root_primitive(feature)
 
-        if not primitive_stack_on_self and isinstance(f_primitive, primitive_class):
-            return False
-
-        if isinstance(f_primitive, tup_primitive_stack_on_exclude):
-            return False
-
-        if feature.number_output_features > 1:
-            return False
-
-        if f_primitive.base_of_exclude is not None and isinstance(
+        # check if stacking is disabled
+        if _check_if_stacking_is_prohibited(
+            feature,
+            f_primitive,
             primitive,
-            tuple(f_primitive.base_of_exclude),
+            primitive_class,
+            primitive_stack_on_self,
+            tuple_primitive_stack_on_exclude,
         ):
             return False
 
-        if primitive_stack_on_self and isinstance(f_primitive, primitive_class):
-            continue
-
-        if tup_primitive_stack_on is None or isinstance(
+        # we permit stacking only if it is not prohibited and meets the criterion to be permitted
+        if _check_if_stacking_is_permitted(
             f_primitive,
-            tup_primitive_stack_on,
+            primitive_class,
+            primitive_stack_on_self,
+            tuple_primitive_stack_on,
         ):
             continue
 
-        if f_primitive.base_of is None:
-            continue
-        if primitive_class in f_primitive.base_of:
-            continue
-
+        # if we reach this case, we default to stacking not being permitted
         return False
```

Comment:
since we continue into a `return False` here we could probably 
just do: `return _check_if_stacking_is_permitted()`

Reply:
We only want to return when `_check_if_stacking_is_permitted` returns `False`, otherwise we need to keep going to make sure it's `True` for all inputs.